docs/dapps/getting-started.md

```diff
@@ -8,91 +8,531 @@ import UntranslatedPageText from "@site/src/components/UntranslatedPageText";
+    }
+
+    const testResult = await TokenFaucet.testWithdrawMethod(testParams)
+    console.log(testResult)
+  })
+})
 ```
+
+A more complex test can be found [here](https://github.com/alephium/nextjs-template/blob/main/test/token.test.ts)
+
+Without entering to much into details, `Typescript` needs some configuration to run the test, just copy the following into `tsconfig.json`
+
+```json
+{
+  "compilerOptions": {
+    "outDir": "dist",
+    "target": "es2020",
+    "esModuleInterop": true,
+    "module": "commonjs",
+    "resolveJsonModule": true
+  },
+  "exclude": ["node_modules"],
+  "include": ["src/**/*.ts", "test/**/*.ts", "scripts/**/*.ts", "alephium.config.ts", "artifacts/**/*.ts"]
+}
+```
+
+You can now run the test:
+
+```sh
 npx @alephium/cli@latest test
 ```
 
+Congratulations, you should see what result would produce a withdraw.
+
 ## Deploy your contract
 
-Next, to deploy the contract we will use Alephium CLI and a deployment script `scripts/0_deploy_faucet.ts`. You can run it using:
+Now things are getting serious, we will deploy our contract on our `devnet` :rocket:
+
+The `deploy` command will deploy everything which is inside the `scripts` folder.
 
+```sh
+mkdir scripts
 ```
+
+Let's create this file into this folder: `scripts/0_deploy_faucet.ts`
+Note that deployment scripts should prefixed with numbers (starting from 0)
+
+```typescript
+import { Deployer, DeployFunction, Network } from '@alephium/cli'
+import { Settings } from '../alephium.config'
+import { TokenFaucet } from '../artifacts/ts'
+
+// This deploy function will be called by cli deployment tool automatically
+// Note that deployment scripts should prefixed with numbers (starting from 0)
+const deployFaucet: DeployFunction<Settings> = async (
+  deployer: Deployer
+): Promise<void> => {
+  const issueTokenAmount = 100n
+  const result = await deployer.deployContract(TokenFaucet, {
+    // The amount of token to be issued
+    issueTokenAmount: issueTokenAmount,
+    // The initial states of the faucet contract
+    initialFields: {
+      symbol: Buffer.from('TF', 'utf8').toString('hex'),
+      name: Buffer.from('TokenFaucet', 'utf8').toString('hex'),
+      decimals: 18n,
+      supply: issueTokenAmount,
+      balance: issueTokenAmount
+    }
+  })
+  console.log('Token faucet contract id: ' + result.contractId)
+  console.log('Token faucet contract address: ' + result.contractAddress)
+}
+
+export default deployFaucet
+```
+
+[deployContract](https://github.com/alephium/alephium-web3/blob/d2b5b63cae015e843aa77b4cf484bc62a070f1d5/packages/cli/src/types.ts#L133-L137) takes our contract and deploy it with the correct arguments, you can also add a `taskTag` arguments, to tag your deployment with a specific name. By default it will use the contract name, but if you deploy multiple time the same contract with different initial fields, your `.deployement` file will get override, using a specific `taskTag` solve this issue.
```

Comment:
```suggestion
The [deployContract](https://github.com/alephium/alephium-web3/blob/d2b5b63cae015e843aa77b4cf484bc62a070f1d5/packages/cli/src/types.ts#L133-L137) of the `Deployer` takes our contract and deploys it with the correct arguments. You can also add a `taskTag` argument to tag your deployment with a specific name. By default, it will use the contract name, but if you deploy the same contract multiple times with different initial fields, your `.deployment` file will get overridden. Using a specific `taskTag` solves this issue.
```